Don't allow insecure, predictable random numbers for IVs

enov opened the pull request:
Silently use MCRYPT_DEV_URANDOM when
the chosen random number generator
is not one of those that are considered secure.

enov committed:
Don't allow insecure, predictable random numbers for IVs

Silently use MCRYPT_DEV_URANDOM when
the chosen random number generator
is not one of those that are considered secure.

enov commented:
This is related to #540 and #545

acoulton commented:
@enov this looks great, but should we document this behaviour on the $_rand property up at the top so it doesn't take people by surprise? Also, is there any reason we can't default that property to MCRYPT_DEV_URANDOM rather than NULL to make the standard behaviour even more obvious?

enov commented:
:+1:

enov committed:
default $_rand to MCRYPT_DEV_URANDOM

and document that we're silently revert to MCRYPT_DEV_URANDOM
when using MCRYPT_RAND

enov commented:
done
